packages/lodestar/src/network/peers/metastore.ts

```diff
@@ -0,0 +1,73 @@
+import {IPeerMetadataStore} from "./interface";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {ReqRespEncoding} from "../../constants";
+import PeerId from "peer-id";
+import {Metadata, Status} from "@chainsafe/lodestar-types";
+import {BasicType, ContainerType} from "@chainsafe/ssz";
+import {StringType} from "./sszTypes";
+
+enum MetadataKey {
+  ENCODING = "encoding",
+  METADATA = "metadata",
+  SCORE = "score",
+  STATUS = "status",
+}
+
+/**
+ * Wrapper around Libp2p.peerstore.metabook
+ * that uses ssz serialization to store data
+ */
+export class Libp2pPeerMetadataStore implements IPeerMetadataStore {
+  private readonly config: IBeaconConfig;
+  private readonly metabook: MetadataBook;
+
+  constructor(config: IBeaconConfig, metabook: MetadataBook) {
+    this.config = config;
+    this.metabook = metabook;
+  }
+
+  public getEncoding(peer: PeerId): ReqRespEncoding | null {
+    return this.get(peer, MetadataKey.ENCODING, new StringType()) as ReqRespEncoding | null;
+  }
+
+  public getMetadata(peer: PeerId): Metadata | null {
+    return this.get(peer, MetadataKey.METADATA, this.config.types.Metadata);
+  }
+
+  public getScore(peer: PeerId): number | null {
+    return this.get(peer, MetadataKey.SCORE, this.config.types.Number64);
+  }
+
+  public getStatus(peer: PeerId): Status | null {
+    return this.get(peer, MetadataKey.STATUS, this.config.types.Status);
+  }
+
+  public setEncoding(peer: PeerId, encoding: ReqRespEncoding): void {
+    return this.set(peer, MetadataKey.ENCODING, new StringType(), encoding);
+  }
+
+  public setMetadata(peer: PeerId, metadata: Metadata): void {
+    return this.set(peer, MetadataKey.METADATA, this.config.types.Metadata, metadata);
+  }
+
+  public setScore(peer: PeerId, score: number): void {
+    return this.set(peer, MetadataKey.SCORE, this.config.types.Number64, score);
+  }
+
+  public setStatus(peer: PeerId, status: Status): void {
+    return this.set(peer, MetadataKey.STATUS, this.config.types.Status, status);
+  }
+
+  private set<T>(peer: PeerId, key: MetadataKey, type: BasicType<T> | ContainerType<T>, value: T): void {
+    this.metabook.set(peer, key, Buffer.from(type.serialize(value)));
```

Comment:
does `MetadataBook` in libp2p 0.28 support Uint8Array?

libp2p 0.29 switches from Buffer to UInt8Array to support browser later so probably we want to use Uint8Array here if possible

Reply:
Buffer is uint8array and deserialize works with uint8array so all of this will work as expected